crates/red_knot_python_semantic/resources/mdtest/narrow/conditionals/nested.md

```diff
@@ -45,3 +45,46 @@ def _(flag1: bool, flag2: bool):
         # TODO should be Never
         reveal_type(x)  # revealed: Literal[1, 2, 3]
 ```
+
+## Cross-scope narrowing
```

Comment:
Nit: our usual "literate test suite" style usually involves breaking up tests into more separate cases than this, with some commentary explaining what functionality each test is supposed to demonstrate, for the benefit of future readers. (Even if this results in a bit more duplication of setting up test context.) It's not a blocker for the PR, but once the PR is otherwise ready I might make an extra pass over this test to break it up a bit.